src/irc.ts

```
@@ -843,13 +849,16 @@ export class Client extends EventEmitter {
         if (message.args[0] !== '+') {
             return;
         }
+        if (!this.opt.saslUsername || !this.opt.saslPassword) {
```

Hmm, I worry this might be a silent error. If we've missed the username or password but expect to use SASL, that feels like a fail. Perhaps we should add this check to the `PLAIN` case statement and throw?